ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -155,12 +155,9 @@ export default Route.extend(UnloadModelRoute, {
     try {
       if (secretModel.failedServerRead) {
         // we couldn't read metadata, so we want to directly fetch the version
-
-        versionModel =
-          this.store.peekRecord('secret-v2-version', JSON.stringify(versionId)) ||
-          (await this.store.findRecord('secret-v2-version', JSON.stringify(versionId), {
-            reload: true,
-          }));
+        versionModel = await this.store.findRecord('secret-v2-version', JSON.stringify(versionId), {
+          reload: true,
+        });
```

Comment:
@Monkeychip I made this change to always fetch the version record to avoid state issues when updating via the CLI which broke this [acceptance test](https://app.circleci.com/pipelines/github/hashicorp/vault/23178/workflows/c453af19-0ccd-4681-bd35-e3a3f0fc1036/jobs/340582) and I wanted to get your take on it.

In the test this policy does not have _read_ permission for the metadata endpoint.
```js
let V2_POLICY_NO_LIST = `
  path "${enginePath}/metadata/*" {
    capabilities = ["update","delete"]
  }
  path "${enginePath}/data/*" {
    capabilities = ["create", "read", "update", "delete"]
  }
`;
```
A new version is created successfully but the subsequent fetch fails. I'm wondering if I need to revert my change, add _read_ to the policy in the test or update the assertion to confirm that the new version does not display after being created?

Reply:
After discussions with Angel and Ivana it was decided to disable the _Create new version_ action when the user does not have read access to the secret metadata which was addressed [in this commit](https://github.com/hashicorp/vault/pull/13166/commits/3aa225ca28d398207e3c750acb9ba76b893e578e). A tooltip is displayed on hover to indicate why the action is not available. I updated the failing acceptance test to assert that the action is now disabled.
![Screenshot from 2021-11-22 14-25-55](https://user-images.githubusercontent.com/24611656/142941765-799eed78-d69e-4b58-a7bb-42fdc9990a6f.png)